server/models/conversationModel.js

```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose");
+
+const Schema = mongoose.Schema;
+
+const conversationSchema = new Schema({
+  participants: [
+    {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User",
+    },
+  ],
+  messages: [
+    {
+      content: {
+        type: String,
+        max: 500, //Max 500 characters
+      },
+      sender: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+      },
+      createdAt: { type: Date },
```

Comment:
I think we don't need to have a separate array for messages. If we just have one schema with fields  participants, content, and other info, everything should work perfectly fine. Everytime you want to fetch the conversations you just need to find all the messages with those participants and you will automatically  have all the messages, there is no need to save them inside a separate array. This way you also will have a simple create function to add new messages, no need to find a conversation and update the messages array. also you don't need to check if there is a conversation and then update that, you can get rid of that code completely.